unison-cli/src/Unison/CommandLine/OutputMessages.hs

```diff
@@ -1131,6 +1132,10 @@ notifyUser dir o = case o of
           else pure mempty
   GitError e -> pure $ case e of
     GitSqliteCodebaseError e -> case e of
+      CodebaseFileLockFailed ->
+        P.wrap $
+          "Failed to obtain a file lock on the codebase. "
+            <> "Perhaps you are running multiple ucm processes against the same codebase."
```

Comment:
For all of these errors maybe mention how to manually clear the lock if you really want to;
This happens to me with my git-index and stack lockfiles sometimes, software messes up and I need to manually clear the lock in order to do anything.

Reply:
>  I would be interested to know what is going on in your git-index and stack examples.

I have no idea how or when this happens, but I've definitely encountered it. I have a general mistrust of any time something is "guaranteed" to happen, ESPECIALLY when it's cross-OS haha.

Reply:
> What happens if you power off the machine — it's in a temp dir that gets cleared on boot?

The file is not deleted. No attempt is ever made to delete the lockfile, and nothing needs to. A file lock is released when the its file descriptor is closed. File locks are an in-memory kernel thing, so it can't survive a reboot.